keep/providers/google_chat_provider/google_provider.py

```diff
@@ -0,0 +1,75 @@
+import os
+import requests
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig
+from keep.contextmanager.contextmanager import ContextManager
+from keep.exceptions.provider_exception import ProviderException
+
+class GoogleChatProvider(BaseProvider):
+    GOOGLE_CHAT_API = "https://chat.googleapis.com/v1/spaces/"
+
+    def __init__(self, context_manager: ContextManager, provider_id: str, config: ProviderConfig):
+        super().__init__(context_manager, provider_id, config)
+
+    def validate_config(self):
+        if not self.config.authentication.get("webhook_url"):
+            raise Exception("Google Chat webhook URL is required")
+
+    def dispose(self):
+        """
+        No need to dispose of anything, so just do nothing.
+        """
+        pass
+
+    def notify(self, message="", cards=[], space="", **kwargs: dict):
+        """
+        Notify a message to a Google Chat room using a webhook URL.
+
+        Args:
+            message (str): The text message to send.
+            cards (list): A list of card objects to attach to the message.
+            space (str): The Google Chat space or room ID.
+
+        Raises:
+            ProviderException: If the message could not be sent successfully.
+        """
+        self.logger.debug("Notifying message to Google Chat")
+        webhook_url = self.config.authentication.get("webhook_url")
+
+        if not message and not cards:
+            raise ProviderException("Message or cards are required")
+
+        if not space:
+            raise ProviderException("Google Chat space (room) is required")
+
+        payload = {
+            "text": message,
+            "cards": cards,
+        }
+
+        response = requests.post(f"{GoogleChatProvider.GOOGLE_CHAT_API}{space}/messages", json=payload)
```

Comment:
Hey @rishi-raj-jain, I think you mistakenly constructed the url instead of using the webhook url that you already extracted from `self.config.authentication` at line 37. Also, the webhook url would already contain the space id so you don't need to mention it separately and more importantly, contains key and token without which you won't be able to send the message in the Google Chat.

Webhook URL would look something like this - https://chat.googleapis.com/v1/spaces/[space_id]/messages?key=[some_key]&token=[some_token]